src/util/address.rs

```diff
@@ -727,6 +727,44 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Returns true if the given pubkey is directly related to the address payload.
+    /// This is determined by directly comparing the address payload with either the
+    /// hash of the given public key or the segwit redeem hash generated from the
+    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
+    pub fn is_related_to_pubkey(&self, pubkey: &PublicKey) -> bool {
+        match self.address_type() {
+            Some(AddressType::P2pkh) | Some(AddressType::P2sh) | Some(AddressType::P2wpkh) | Some(AddressType::P2wsh) => {
+                let pubkey_hash = pubkey.pubkey_hash();
+                let payload = self.payload_as_bytes();
+
+                (*pubkey_hash == *payload) || (*segwit_redeem_hash(&pubkey_hash) == *payload)
+            },
+            Some(AddressType::P2tr) => {
+                let xonly_pubkey = XOnlyPublicKey::from(pubkey.inner);
```

Comment:
Is this really a good idea if the parity doesn't match?

Reply:
I just dropped this entirely